lib/ocpp/v201/device_model_storage_sqlite.cpp

```diff
@@ -14,17 +18,33 @@ namespace v201 {
 extern void filter_criteria_monitors(const std::vector<MonitoringCriterionEnum>& criteria,
                                      std::vector<VariableMonitoringMeta>& monitors);
 
-DeviceModelStorageSqlite::DeviceModelStorageSqlite(const fs::path& db_path) {
+DeviceModelStorageSqlite::DeviceModelStorageSqlite(const fs::path& db_path, const fs::path& migration_files_path,
+                                                   const fs::path& schemas_path, const fs::path& config_path,
+                                                   const bool init_db) {
+    if (init_db) {
+        if (db_path.empty() || migration_files_path.empty() || schemas_path.empty() || config_path.empty()) {
+            EVLOG_AND_THROW(
+                DeviceModelStorageError("Can not initialize device model storage: one of the paths is empty."));
+        }
+        InitDeviceModelDb init_device_model_db(db_path, migration_files_path);
+        init_device_model_db.initialize_database(schemas_path, false);
+        init_device_model_db.insert_config_and_default_values(schemas_path, config_path);
+    }
```

Comment:
Yes that constructor thing is a bit strange indeed
Problem is that I have to do the device model initialization before creating DeviceModelStorageSqlite.

Reply:
I removed the odd looking constructor and added default parameters to the remaining constructor. Because we do so much code in the constructors, I don't know how else to do it, because like I said the code has to be called before the code in the chargepoint constructor is called, which creates a device model instance. But all the migration and inserting stuff should have been done at that point already.